luxonis_train/utils/general.py

```
@@ -152,14 +157,37 @@ def get_with_default(
+    exclude = set(exclude or [])
+    exclude |= {"self", "kwargs"}
+    signature = dict(inspect.signature(func).parameters)
+    return {
+        name: param for name, param in signature.items() if name not in exclude
+    }
+
+
 def safe_download(
     url: PathType | None,
     file: str | None = None,
     dir: PathType = ".cache/luxonis_train",
```

Ruff (0.15.13)

[error] Function argument `dir` is shadowing a Python builtin

(A002)